server/tests/fstar_connection.test.ts

```diff
@@ -0,0 +1,100 @@
+import { beforeEach, describe, expect, test, jest } from '@jest/globals';
+
+import { FStarConnection } from '../src/fstar_connection';
+
+
+describe('bufferedMessageHandler tests', () => {
+	// The function we'll be testing, reset before each test
+	let bufferedMessageHandler: (msg: string) => void;
+	const mockHandler = jest.fn();
+
+	beforeEach(() => {
+		jest.clearAllMocks();
+
+		// Reset the function we're testing, this will create a fresh buffer.
+		bufferedMessageHandler = FStarConnection.bufferedMessageHandlerFactory(mockHandler);
+	});
+
+	test('test valid message', () => {
+		const valid_message = '{"kind": "message"}';
```

Comment:
If you want to keep these tests, then we should at least test behaviors that can actually happen.  Suggestion for test cases:
- `['{"kind": "message"}\n']`
- `['{"kind": ', '"message"}\n']` (fragmented)
- `['{"kind": ', '"message"}\n{"kind": ', '"message"}\n']` (combined and fragmented)

Note the newline after each message.  There are also no newlines in any message.

Reply:
At this point I think I should do the refactoring in a separate PR that can be merged quickly. I opened #25 to track this issue.